designs/NuGetExtensibility/INuGetProjectServices.md

```diff
@@ -0,0 +1,122 @@
+# INuGetProjectServices
+
+* Status: In Review
+* Author(s): [Andy Zivkovic](https://github.com/zivkan)
+* Issue: [1](https://github.com/NuGet/Home/issues/9577) IVsPackageInstallerServices should be async
+
+## Problem Background
+
+The existing IVsPackageInstallerServices service is synchronous, and is being called by several components on the UI thread, causing UI delays in Visual Studio. We are creating new async APIs that will allow us to unblock the UI thread, and use the Visual Studio Service Broker to work in Visual Studio CodeSpaces (and maybe Live Share).
```

Comment:
Are you going to replace the inner bits of NuGet IVs* with these async interfaces/implementations to avoid code duplication ? Or, Are IVs* and this interface for different purposes?

Reply:
We never use our own IVs* interfaces. 

They are only used for extensibility purposes.

Reply:
This is an implementation question, whereas this document is the API design. Resolving as out of scope (but it's a very relevant question for the PR that implements this API).